File: 404-sum-of-left-leaves/sum-of-left-leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        int ans =0 ; 
        if (!root )return ans; 
        
        // concept : level order traversal 
        
        queue<pair<TreeNode*, bool >> q; 
        q.push(make_pair(root , false )); 
        
        while (!q.empty()){
           
           pair<TreeNode*  , bool > frontValue = q.front() ; 
           q.pop(); 
           
           TreeNode* frontNode = frontValue.first ; 
           bool check = frontValue.second ; 
           
           if (check == true and (frontNode->left== NULL and frontNode->right == NULL )){
               ans += frontNode->val ;
           }
           
           
           if(frontNode->left ){
               q.push(make_pair(frontNode->left , true )); 
           }
           
           if (frontNode->right ){
               q.push(make_pair(frontNode->right , false ));
           }
        }
        
        return ans;
    }
};

File: 404-sum-of-left-leaves/sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        // level order traversal 
        if (root  == null )return 0 ; 
        int ans=0 ;

        Queue<TreeNode> q = new LinkedList<>() ; 
        q.offer(root ); 


        while (!q.isEmpty()){
            TreeNode frontNode = q.peek()  ; 
            q.poll(); 

            if (frontNode.left!= null && (frontNode.left.left == null && frontNode.left.right == null )) {          
                ans += (frontNode.left.val);
            }


            if (frontNode.left != null ){
                q.offer(frontNode.left);
            }

            if (frontNode.right != null){
                q.offer(frontNode.right);
            }
        }

        return ans;
    }
}